### DISCUSSION TITLE
Async configuration of server failed

### DISCUSSION THREAD

**Discussion by Smokyr** [fail2ban/fail2ban#3672] [Q&A] [answered]
Hi,

My fail2ban server seems to work well, but when i send the command "service fail2ban status", server is running:

● fail2ban.service - Fail2Ban Service
   Loaded: loaded (/etc/systemd/system/fail2ban.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2024-01-31 19:05:46 CET; 2s ago
     Docs: man:fail2ban(1)
  Process: 7786 ExecStartPre=/bin/mkdir -p /var/run/fail2ban (code=exited, status=0/SUCCESS)
 Main PID: 25173 (fail2ban-server)
    Tasks: 0 (limit: 4915)
   Memory: 320.0K
   CGroup: /system.slice/fail2ban.service
           ‣ 25173 /usr/bin/python3 /usr/bin/fail2ban-server --async -b -s /var/run/fail2ban/fail2ban.sock -p /var/run/fail2ban/fail2ban.pid --loglevel

Jan 31 19:05:46 myserver systemd[1]: Starting Fail2Ban Service...
Jan 31 19:05:46 myserver systemd[1]: Started Fail2Ban Service.
Jan 31 19:05:46 myserver fail2ban-server[7787]:  Server already running
Jan 31 19:05:46 myserver fail2ban-server[7787]:  Async configuration of server failed

What means the last line ?

**Comment by sebres** [comment, marked as answer]
Hmm. That is strange... Normally it'd generate the errors in configuration in the same journal. Anyway...

The message "Server already running" may signal that there is another fail2ban instance (found by the pid-file).
Maybe systemd-unit lost a state of process for some reasons.
Anyway you have to stop it firstly, e. g. `?sudo? fail2ban-client stop`, or kill and delete the pid-file (however in that it can leave some chains in the net-filter in the inappropriate state.

"Async configuration of server failed" means the errors in configs, what exactly is wrong you can also see with:
```
?sudo? fail2ban-client -vv -t
```

> `/usr/bin/python3 /usr/bin/fail2ban-server --async -b -s ...`

Your unit service-file seems to be different to our stock version - here [0.11](https://github.com/fail2ban/fail2ban/blob/0.11.2/files/fail2ban.service.in) and [1.0](https://github.com/fail2ban/fail2ban/blob/1.0.2/files/fail2ban.service.in).
I checked several releases and `ExecStart` is always `fail2ban-server -xf start` and pid-file is specified via `PIDFile`.
Which platform and fail2ban version do you use and how fail2ban is installed?

**Reply by Smokyr** [reply]
Hi,

fail2ban-client -vv -t give:
OK: configuration test is successful

for "The message "Server already running" may signal that there is another fail2ban instance", it was my fault as I try to start it twice 

I am on Debian 4.19.98-1+deb10u1, and the version of fail2ban is  v0.10.2

**Reply by Smokyr** [reply]
Ok, that was my fault: I stop fail2ban and start it again and:
fail2ban.service - Fail2Ban Service
   Loaded: loaded (/etc/systemd/system/fail2ban.service; enabled; vendor preset: enabled)
   Active: active (running) since Thu 2024-02-01 12:44:59 CET; 4s ago
     Docs: man:fail2ban(1)
  Process: 7994 ExecStartPre=/bin/mkdir -p /var/run/fail2ban (code=exited, status=0/SUCCESS)
 Main PID: 7995 (fail2ban-server)
    Tasks: 61 (limit: 4915)
   Memory: 25.3M
   CGroup: /system.slice/fail2ban.service
           └─7995 /usr/bin/python3 /usr/bin/fail2ban-server -xf start

Feb 01 12:44:59 myserver systemd[1]: Starting Fail2Ban Service...
Feb 01 12:44:59 myserver systemd[1]: Started Fail2Ban Service.
Feb 01 12:45:00 myserver fail2ban-server[7995]: Server ready

So no more this message.

Sorry for the time spent for me
